src/de.rs

```diff
@@ -243,6 +243,48 @@ impl<'de, R: Read<'de>> Deserializer<R> {
         Error::syntax(reason, position.line, position.column)
     }
 
+    /// Returns the first non-comment byte without consuming it, or `None` if
+    /// EOF is encountered.
+    fn parse_comment(&mut self) -> Result<Option<u8>> {
+        // Cursor is already after the first slash.
+        match tri!(self.peek()) {
+            // Line Comment
+            Some(b'/') => {
+                self.eat_char();
+
+                // And eat the rest of the line.
+                loop {
+                    match tri!(self.peek()) {
+                        // Found the end of the line.
+                        // Pass it back to parse_whitespace.
+                        Some(b'\r') | Some(b'\n') => {
+                            self.eat_char();
+                            return self.parse_whitespace()
+                        },
+                        _ => {
+                            self.eat_char();
+                        }
+                    };
+                }
+            }
+
+            // Block Comment
+            Some(b'*') => {
+                self.eat_char();
+
+                // TODO: eat until */.
+                return Ok(Some(b'/'));                
+            }
+
+            // Random bad `/` in a whitespace position.
+            _ => {
+                // Lie, false rewind.
+                // FIXME: Actually rewind?
+                return Ok(Some(b'/'));
```

Comment:
Should we error here instead? `parse_whitespace` is supposed to return the first non-whitespace without consuming it, but this codepath is returning the first non-whitespace *and* consuming it.

Reply:
Yeah, I know this is cheating. I think the right answer here is to change where we eat the character and peek a second `u8` when necessary.